### ISSUE TITLE
Failed to get migration status: 'MigrationStatus' object has no attribute 'info'

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#353]
Originally reported of Google Code with ID 297.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".
gnt-cluster (ganeti v2.6.0) 2.6.0
Software version: 2.6.0
Internode protocol: 2060000
Configuration format: 2060000
OS api version: 20
Export interface: 0

<b>What distribution are you using?</b>
Gentoo

<b>What steps will reproduce the problem?</b>
1. gnt-instance migrate GUEST

<b>What is the expected output? What do you see instead?</b>
Mostly work, but sometimes I get this error message.

<b>Please provide any additional information below.</b>

# gnt-instance migrate GUEST
Instance GUEST will be migrated. Note that migration
might impact the instance if anything goes wrong (e.g. due to bugs in
the hypervisor). Continue?
y/[n]/?: y
Thu Nov  1 17:44:15 2012 Migrating instance GUEST
Thu Nov  1 17:44:15 2012 * checking disk consistency between source and target
Thu Nov  1 17:44:18 2012 * switching node node1.cluster2.ganeti.dmz to secondary mode
Thu Nov  1 17:44:19 2012 * changing into standalone mode
Thu Nov  1 17:44:20 2012 * changing disks into dual-master mode
Thu Nov  1 17:44:22 2012 * wait until resync is done
Thu Nov  1 17:44:23 2012 * preparing node1.cluster2.ganeti.dmz to accept the instance
Thu Nov  1 17:44:24 2012 * migrating instance to node1.cluster2.ganeti.dmz
Thu Nov  1 17:44:25 2012 * starting memory transfer
Thu Nov  1 17:44:35 2012 * memory transfer progress: 31.62 %
Thu Nov  1 17:44:46 2012 * memory transfer progress: 70.70 %
Thu Nov  1 17:45:00 2012 * memory transfer progress: 83.96 %
Thu Nov  1 17:45:21 2012 Migration failed, aborting
Thu Nov  1 17:45:26 2012 * switching node node1.cluster2.ganeti.dmz to secondary mode
Thu Nov  1 17:45:28 2012 * changing into standalone mode
Thu Nov  1 17:45:30 2012 * changing disks into single-master mode
Thu Nov  1 17:45:32 2012 * wait until resync is done
Thu Nov  1 17:45:35 2012    - progress: 0.0%
Failure: command execution error:
Could not migrate instance GUEST: Failed to get migration status: 'MigrationStatus' object has no attribute 'info'
```
Originally added on 2012-11-02 14:57:25 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-11-07 14:05:56 +0000 UTC.

Added Labels: Type-Defect
Added to Milestone: Release2.6

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 62457f5
Author: Iustin Pop <iustin@google.com>
Date:   Mon Nov 12 12:50:52 2012 +0100

    Fix type error in kvm/GetMigrationStatus
    
    Commit 6a1434d7 (“Make migration RPC non-blocking”) changed from
    raising HypervisorErrors to returning MigrationStatus
    objects. However, these objects don't have an "info" attribute, so
    they can't pass a reason back (which is in itself a bug); but the KVM
    hypervisor code attempts to do so, and fails at runtime with:
    
      Failed to get migration status: 'MigrationStatus' object has no attribute 'info'
    
    instead of the intended:
    
      Migration failed, aborting: too many broken 'info migrate' answers
    
    For now (on stable-2.6), let's just remove the "info" reason, and
    later we can add it back properly once we have a way to correctly
    represent migration status failures in the LU.
    
    This fixes issue 297.
    
    Signed-off-by: Iustin Pop <iustin@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>

```
Originally added on 2012-12-12 10:14:07 +0000 UTC.

Mentioned Issue with GoogleCode ID 297 has been migrated to GitHub with ID #353.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking as released the bugs fixed in 2.6.2.
```
Originally added on 2013-01-11 12:13:05 +0000 UTC.

Changed State: Released
